notebooks/225-yolov7-optimization/225-yolov7-optimiization.ipynb

```diff
@@ -0,0 +1,821 @@
+   "source": [
+    "from PIL import Image\n",
+    "# visualize prediction result\n",
+    "Image.open('runs/detect/exp/horses.jpg')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Export to ONNX"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To export and ONNX format of the model, we will use `export.py` script. Let's check it's arguments."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!python export.py --help"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The most important parameters:\n",
+    "* `--weights` - path to model weigths checkpoint\n",
+    "* `--img-size` - size of input image for onnx tracing\n",
+    "\n",
+    "When exporting the ONNX model from PyTorch, there is an opportunity to setup configurable parameters for including post-processing results in model:\n",
+    "* `--end2end` - export full model to onnx including postporcessing\n",
+    "* `--grid` - export Detect layer as part of model\n",
+    "* `--topk-all` - topk elements for all images\n",
+    "* `--iou-thres` - intersection over union threshold for NMS\n",
+    "* `--conf-thres` - minimal confidence threshold\n",
+    "* `--max-wh` - max bounding box width and height for NMS\n",
+    "\n",
+    "Including whole postprocessing to model can help to achieve more perfomant resuls, but in the same time it makes model less flaxible and does not guarantee full accuracy reproducibility. \n",
+    "It is the reason why we will add only `--grid` parameter to preserve original pytorch model result format.\n",
+    "If you want to understand how to work with end2end ONNX model, you can check this [notebook](https://github.com/WongKinYiu/yolov7/blob/main/tools/YOLOv7onnx.ipynb)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!python export.py --weights model/yolov7-tiny.pt --grid"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Convert ONNX Model to OpenVINO Intermideate Representation (IR)\n",
+    "While ONNX models are directly supported by OpenVINO runtime, it can be useful to convert them to IR format to take advantage of OpenVINO optimization tools and features.\n",
+    "`mo.convert` python function can be used for converting model using OpenVINO Model Optimizer. \n",
+    "The function returns instance of OpenVINO Model class, which is ready to use in Python interface but can also be serialized to OpenVINO IR format for future execution."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from openvino.tools import mo\n",
+    "from openvino.runtime import serialize\n",
+    "\n",
+    "model = mo.convert(input_model='model/yolov7-tiny.onnx')\n",
```

Comment:
In the recent master, we renamed it to `convert_model`, and you don't need to specify `input_model` anymore (but you still can if you want).

Reply:
I still waiting package which will include this changes, after that I update...